lib/elixir/lib/enum.ex

```diff
@@ -1408,7 +1408,14 @@ defmodule Enum do
 
   """
   @spec sum(t) :: number
-  def sum(enumerable) do
+  def sum(first..last) when last > first do
+    ((last*(last+1))/2) - ((first*(first-1))/2)    
+  end
+  def sum(first..last) do # When first >= last
+    ((first*(first+1))/2) - ((last*(last-1))/2)    
```

Comment:
I'd even do `def sum(first..last), do: sum(last..first)` here.

In the current version one has to maintain two instances of the same formula.